packages/terra-navigation-side-menu/src/NavigationSideMenu.jsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import List from 'terra-list';
+import ActionHeader from 'terra-clinical-action-header';
+import ContentContainer from 'terra-content-container';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+
+import styles from './NavigationSideMenu.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+};
+
+const propTypes = {
+  /**
+   * An array of configuration for each menu item.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     * Keys of menu items
+     */
+    children: PropTypes.arrayOf(PropTypes.string),
+    /**
+     * ID to be applied to the menu item div.
+     */
+    id: PropTypes.string,
+    /**
+     * Unique identifier that will be returned in the onChange callback when an endpoint is reached.
+     */
+    key: PropTypes.string.isRequired,
+    /**
+     * Title for the menu row and header title when selected.
+     */
+    title: PropTypes.string.isRequired,
+  })),
+  /**
+   * Callback function when a menu endpoint is reached.
+   */
+  onChange: PropTypes.func.isRequired,
+  /**
+   * Delegate prop that is added by the NavigationLayout.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType,
+  /**
+   * Key of the top level menu.
+   */
+  initialSelectedKey: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  menuItems: [],
+};
+
+const processMenuItems = (items) => {
+  const processedItems = {};
+  items.forEach((item) => {
+    processedItems[item.key] = { id: item.id, title: item.title, children: item.children };
+  });
+  return processedItems;
+};
+
+class NavigationSideMenu extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.onBackClick = this.onBackClick.bind(this);
+    this.onItemClick = this.onItemClick.bind(this);
+
+    this.state = { processedItems: processMenuItems(props.menuItems), selectedKey: props.initialSelectedKey, selectedChildKey: null };
+    this.previousStack = [];
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({
+      processedItems: processMenuItems(nextProps.menuItems),
+    });
+  }
+
+  onBackClick() {
+    if (this.previousStack.length) {
+      this.setState({ selectedKey: this.previousStack.pop(), selectedChildKey: this.state.selectedKey });
+    }
+  }
+
+  onItemClick(event, key) {
+    const clickedItem = this.state.processedItems[key];
+    if (clickedItem.children && clickedItem.children.length) {
+      this.previousStack.push(this.state.selectedKey);
+      this.setState({ selectedKey: key, selectedChildKey: key });
+    } else {
+      this.props.onChange(event, key);
+      this.setState({ selectedChildKey: key });
+    }
+  }
+
+  buildListItem(key) {
+    const item = this.state.processedItems[key];
+    return (
+      <List.Item
```

Comment:
Does the side-menu support sub-navigation at all?

Reply:
I know you once mentioned sub nav would be multiple side navs stacked on top of one another. If this is the expected implementation here, can we add a way to include some sort of sub-nav indicator?

Reply:
This now support sub navigation based on config. I updated the code to make this no longer a single component, it's a menu stack by default.

Reply:
I'd be fine adding some sort of indicator, right now I don't have one as it matches the app links. We'd just need to try to keep them matching a similar design.